frontend/packages/dev-console/src/components/pipelines/detail-page-tabs/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { PipelineParam } from '@console/dev-console/src/utils/pipeline-augment';
+import { pipelineParameters } from './utils-data';
+import { removeEmptyDefaultFromPipelineParams } from '../utils';
+
+describe('removeEmptyDefaultFromPipelineParams omits empty default values', () => {
+  it('should return pipline parameters by omitting empty default values', () => {
```

Comment:
When writing tests, you should always consider "what can the user of this function provide".

- `null` **value?** - sometimes we can consider this an invalid use of the function and not support the test, just depends
- **empty value** (in this case `[]`) - tests the function doesn't do anything unexpected
- **a value that does NOT need to be modified** - sometimes values are accidentally modified and thus this test helps know that things shouldn't always be touched
- **a value that DOES need to be modified** - this is what you have here, this is a good test to showcase it does do what it does

Essentially you should always look at writing tests that cement the current functionality. Sure, we know your function only modifies the field on the 2nd element, and sure we know that the first and third elements are unmodified thus showing it doesn't touch them... but the goal of tests is to guard against accidental "breaks" or bugs, which include paths that don't do anything.

Please add a few more tests, consider any other path you think you might want to test (those without `default`s?)... rarely is 1 test good enough to say a function is fully tested.

Reply:
Every `removeEmptyDefaultFromPipelineParams` call deserves it's own `it` wrapper... Debugging tests can be quite annoying if they are not broken up.